### PR TITLE
Improve the error message

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java [hunk 1]
```diff
@@ -266,7 +266,11 @@ public static Object copy(BString sourcePath, BString destinationPath, BString..
                 Files.walkFileTree(srcPath, new RecursiveFileCopyVisitor(srcPath, destPath, ops));
             }
         } catch (IOException ex) {
-            return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR, ex);
+            return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR,
+                    "An error occurred when copying the file/s: " + ex);
+        } catch (NullPointerException ex) {
+            return FileUtils.getBallerinaError(FileConstants.FILE_NOT_FOUND_ERROR,
+                    "The target directory does not exist: " + ex);
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java:273]: Do we need ex in the error message? can we avoid this null pointer exception

Comment (kalaiyarasiganeshalingam) [reply native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java]: Avoided in [c2f3749](https://github.com/ballerina-platform/module-ballerina-file/pull/430/commits/c2f374913d09e40da0b7ea6227560a6cd5d7903a)

File: native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java [hunk 2]
```diff
@@ -266,7 +266,12 @@ public static Object copy(BString sourcePath, BString destinationPath, BString..
                 Files.walkFileTree(srcPath, new RecursiveFileCopyVisitor(srcPath, destPath, ops));
             }
         } catch (IOException ex) {
-            return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR, ex);
+            if (ex.toString().contains("NoSuchFileException")) {
+                return FileUtils.getBallerinaError(FileConstants.FILE_NOT_FOUND_ERROR,
+                        "The target directory does not exist: " + ex.getMessage());
+            }
+            return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR,
+                    "An error occurred when copying the file/s: " + ex.getMessage());
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java:274]: } catch (NoSuchFileException ex) {
        return FileUtils.getBallerinaError(FileConstants.FILE_NOT_FOUND_ERROR,
                        "The target directory does not exist: " + ex.getMessage());
} catch (IOException ex) {
        return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR,
                    "An error occurred when copying the file/s: " + ex.getMessage());
}

Comment (daneshk) [reply native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java:274]: Shall we do like this

Comment (kalaiyarasiganeshalingam) [reply native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java:273]: Updated.

**Review by daneshk — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**